lib/sqlalchemy/util/langhelpers.py

```diff
@@ -1470,14 +1470,14 @@ def __eq__(self, other):
         return hash(self) == hash(other)
 
 
-def warn(msg):
+def warn(msg, stacklevel=2):
     """Issue a warning.
 
     If msg is a string, :class:`.exc.SAWarning` is used as
     the category.
 
     """
-    warnings.warn(msg, exc.SAWarning, stacklevel=2)
+    warnings.warn(msg, exc.SAWarning, stacklevel=stacklevel)
```

Comment:
I'm not using this now so it can be reverted.

I should have probably started from zero, instead of updating the previous implementation

Reply:
i've had only bad experiences trying to get stacklevel to be useful

Reply:
i'll remove the change